subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/ModuleResolveState.java

```diff
@@ -291,4 +293,21 @@ VirtualPlatformState getPlatformState() {
         }
         return platformState;
     }
+
+    void removeHardEdge() {
```

Comment:
:x: Now that the methods are on `ModuleResolveState` I think it can be confusing to have a method named "remove hard edge" or "add hard edge", because this shouldn't deal with edges. Maybe methods like "incrementHardEdgeCount" would make more sense, WDYT?